### PR TITLE
[1.16] base-images: Use debian-base:v2.1.0 and debian-iptables:v12.1.0 (includes CVE fixes)

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/sig release
/area release-eng dependency security

**What this PR does / why we need it**:

- Update dependents to use `debian-base:v2.1.0`
- Update dependents to use `debian-iptables:v12.1.0`

(Selective cherry pick of https://github.com/kubernetes/kubernetes/pull/90665, https://github.com/kubernetes/kubernetes/pull/90697, and https://github.com/kubernetes/kubernetes/pull/90782.)

/assign @dims @BenTheElder
cc: @kubernetes/release-engineering
/priority important-soon

**Which issue(s) this PR fixes**:

Tracking issue: https://github.com/kubernetes/kubernetes/issues/58012

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- base-images: Use debian-base:v2.1.0 (update to Debian Buster, includes CVE fixes)
- base-images: Use debian-iptables:v12.1.0 (update to Debian Buster, includes CVE fixes and iptables-nft/iptables-legacy wrapper)
```

### CODE REVIEW

**Comment by justaugustus** [comment]
Thank @dims!
For `build/`:
/assign @cblecker @fejta 

For `cluster/`:
/assign @mikedanese @spiffxp

File: build/common.sh
Comment (cblecker) [inline build/common.sh:96]: We are bumping a major version on 1.16.. will that make a difference here?

Comment (justaugustus) [reply build/common.sh:96]: @cblecker -- Should be fine, but I can run a mock stage after this merges to validate. I think this was more so the bump not getting back to the 1.16 branch in whichever v2.0.0/v12.0.0 update vs a specific reason the base image shouldn't be kept up-to-date.

Comment (BenTheElder) [reply build/common.sh:96]: I'd like to actually inspect the delta.

one thing is that we're backporting the iptables / iptables nft "fix", which we previously suggested we wouldn't,

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by fejta** [comment]
Ben and Christoph seem on top of this one
/uncc
/unassign

**Comment by BenTheElder** [comment]
particularly notable:
debian-base 1.0.1 => 2.1.0 includes a bump to buster https://github.com/kubernetes/kubernetes/pull/82966
debian-iptables 11.0.3 => 12.1.0 includes a bump to the magic iptables-nft / iptables-legacy wrapper (probably desirable) in addition to the bump to buster

these might be worth a more detailed release note, at least ..?

**Comment by BenTheElder** [comment]
previous notes in https://github.com/kubernetes/kubernetes/pull/82966

**Comment by tpepper** [comment]
Since this combines cleanup and image version bumping (CVE fixes), I think it would be good to add something like a "(CVE fixes)" parenthetical to the release note.

**Comment by justaugustus** [comment]
> Since this combines cleanup and image version bumping (CVE fixes), I think it would be good to add something like a "(CVE fixes)" parenthetical to the release note.

@tpepper -- Done!

Needs re-lgtm after rebasing in the [kube-cross bump](https://github.com/kubernetes/release/issues/1281).

**Comment by justaugustus** [comment]
Apt package diffs:

```
$ container-diff diff k8s.gcr.io/debian-base-amd64:v1.0.1 us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0 --type=apt 

-----Apt-----

Packages found only in k8s.gcr.io/debian-base-amd64:v1.0.1:
NAME                      VERSION                      SIZE
-gcc-6-base               6.3.0-18 deb9u1              209K
-iproute2                 4.9.0-1 deb9u1               1.7M
-iputils-ping             3:20161105-1                 111K
-libcap2                  1:2.25-1                     47K
-libcomerr2               1.43.4-2 deb9u1              83K
-libelf1                  0.168-1                      934K
-libidn11                 1.33-1 deb9u1                307K
-libmnl0                  1.0.4-2                      46K
-libtinfo5                6.0 20161126-1+deb9u2        478K
-libustr-1.0-1            1.0.4-6                      258K
-libuuid1                 2.29.2-1 deb9u1              107K
-lsb-base                 9.20161125                   49K
-multiarch-support        2.24-11 deb9u4               221K
-sensible-utils           0.0.9 deb9u1                 62K

Packages found only in us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0:
NAME                  VERSION                SIZE
-bsdutils             1:2.33.1-0.1           293K
-gcc-8-base           8.3.0-6                248K
-libffi6              3.2.1-9                56K
-libgmp10             2:6.1.2 dfsg-4         565K
-libgnutls30          3.6.7-4 deb10u3        2.6M
-libhogweed4          3.4.1-1                245K
-libidn2-0            2.0.5-1 deb10u1        280K
-libp11-kit0          0.23.15-2              1.3M
-libseccomp2          2.3.3-4                306K
-libsystemd0          241-7~deb10u3          767K
-libtasn1-6           4.13-3                 112K
-libudev1             241-7~deb10u3          258K
-libunistring2        0.9.10-1               1.6M
-libzstd1             1.3.8 dfsg-3           670K

Version differences:
PACKAGE                        IMAGE1 (k8s.gcr.io/debian-base-amd64:v1.0.1)        IMAGE2 (us.gcr.io/k8s-artifacts-prod/build-image/debian-base-amd64:v2.1.0)
-adduser                       3.115, 849K                                         3.118, 849K
-apt                           1.4.9, 3.5M                                         1.8.2, 4M
-base-files                    9.9 deb9u12, 333K                                   10.3 deb10u3, 340K
-base-passwd                   3.5.43, 229K                                        3.5.46, 232K
-coreutils                     8.26-3, 14.7M                                       8.30-3, 15.4M
-dash                          0.5.8-2.4, 204K                                     0.5.10.2-5, 212K
-debconf                       1.5.61, 558K                                        1.5.71, 520K
-debian-archive-keyring        2017.5 deb9u1, 148K                                 2019.1, 198K
-debianutils                   4.8.1.1, 213K                                       4.8.6.1, 226K
-diffutils                     1:3.5-3, 1.3M                                       1:3.7-3, 1.5M
-dpkg                          1.18.25, 6.6M                                       1.19.7, 6.5M
-findutils                     4.6.0 git+20161106-2, 1.8M                          4.6.0 git+20190209-2, 1.9M
-gpgv                          2.1.18-8~deb9u4, 721K                               2.2.12-1 deb10u1, 837K
-grep                          2.27-2, 1.1M                                        3.3-1, 1014K
-gzip                          1.6-5 b1, 231K                                      1.9-3, 243K
-hostname                      3.18 b1, 47K                                        3.21, 54K
-init-system-helpers           1.48, 131K                                          1.56 nmu1, 133K
-libacl1                       2.2.52-3 b1, 62K                                    2.2.53-4, 70K
-libapt-pkg5.0                 1.4.9, 3M                                           1.8.2, 3.2M
-libattr1                      1:2.4.47-2 b2, 42K                                  1:2.4.48-4, 55K
-libaudit-common               1:2.6.7-2, 30K                                      1:2.8.4-3, 33K
-libaudit1                     1:2.6.7-2, 150K                                     1:2.8.4-3, 161K
-libbz2-1.0                    1.0.6-8.1, 96K                                      1.0.6-9.2~deb10u1, 104K
-libc-bin                      2.24-11 deb9u4, 3.3M                                2.28-10, 3.4M
-libc6                         2.24-11 deb9u4, 10.4M                               2.28-10, 12M
-libcap-ng0                    0.7.7-3 b1, 43K                                     0.7.9-2, 47K
-libdb5.3                      5.3.28-12 deb9u1, 1.8M                              5.3.28 dfsg1-0.5, 1.8M
-libdebconfclient0             0.227, 67K                                          0.249, 72K
-libgcc1                       1:6.3.0-18 deb9u1, 108K                             1:8.3.0-6, 116K
-libgcrypt20                   1.7.6-2 deb9u3, 1.2M                                1.8.4-5, 1.3M
-libgpg-error0                 1.26-2, 572K                                        1.35-1, 175K
-liblz4-1                      0.0~r131-2 b1, 93K                                  1.8.3-1, 141K
-liblzma5                      5.2.2-1.2 b1, 339K                                  5.2.4-1, 263K
-libnettle6                    3.3-1 b2, 358K                                      3.4.1-1, 380K
-libpam-modules                1.1.8-3.6, 874K                                     1.3.1-5, 1M
-libpam-modules-bin            1.1.8-3.6, 220K                                     1.3.1-5, 238K
-libpam-runtime                1.1.8-3.6, 1016K                                    1.3.1-5, 1M
-libpam0g                      1.1.8-3.6, 229K                                     1.3.1-5, 244K
-libpcre3                      2:8.39-3, 668K                                      2:8.39-12, 673K
-libselinux1                   2.6-3 b3, 209K                                      2.8-1 b1, 194K
-libsemanage-common            2.6-2, 39K                                          2.8-2, 30K
-libsemanage1                  2.6-2, 291K                                         2.8-2, 301K
-libsepol1                     2.6-2, 653K                                         2.8-1, 724K
-libstdc++6                    6.3.0-18 deb9u1, 2M                                 8.3.0-6, 2M
-login                         1:4.4-4.1, 2.7M                                     1:4.5-1.1, 2.6M
-passwd                        1:4.4-4.1, 2.4M                                     1:4.5-1.1, 2.5M
-perl-base                     5.24.1-3 deb9u6, 7.4M                               5.28.1-6, 9.9M
-sed                           4.4-1, 799K                                         4.7-1, 883K
-tar                           1.29b-1.1, 2.7M                                     1.30 dfsg-6, 2.8M
-zlib1g                        1:1.2.8.dfsg-5, 156K                                1:1.2.11.dfsg-1, 174K


$ container-diff diff k8s.gcr.io/debian-iptables-amd64:v11.0.3 us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.1.0 --type=apt 

-----Apt-----

Packages found only in k8s.gcr.io/debian-iptables-amd64:v11.0.3:
NAME                      VERSION                         SIZE
-gcc-6-base               6.3.0-18 deb9u1                 209K
-iproute2                 4.9.0-1 deb9u1                  1.7M
-iputils-ping             3:20161105-1                    111K
-libcap2                  1:2.25-1                        47K
-libcomerr2               1.43.4-2 deb9u1                 83K
-libelf1                  0.168-1                         934K
-libidn11                 1.33-1 deb9u1                   307K
-libip4tc0                1.6.0 snapshot20161117-6        110K
-libip6tc0                1.6.0 snapshot20161117-6        114K
-libipset3                6.30-2                          289K
-libtinfo5                6.0 20161126-1+deb9u2           478K
-libustr-1.0-1            1.0.4-6                         258K
-libuuid1                 2.29.2-1 deb9u1                 107K
-multiarch-support        2.24-11 deb9u4                  221K
-sensible-utils           0.0.9 deb9u1                    62K

Packages found only in us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.1.0:
NAME                  VERSION                 SIZE
-bsdutils             1:2.33.1-0.1            293K
-gcc-8-base           8.3.0-6                 248K
-libffi6              3.2.1-9                 56K
-libgmp10             2:6.1.2 dfsg-4          565K
-libgnutls30          3.6.7-4 deb10u3         2.6M
-libhogweed4          3.4.1-1                 245K
-libidn2-0            2.0.5-1 deb10u1         280K
-libip4tc2            1.8.3-2~bpo10 1         119K
-libip6tc2            1.8.3-2~bpo10 1         119K
-libipset11           6.38-1.2                207K
-libnftnl11           1.1.5-1~bpo10 1         221K
-libp11-kit0          0.23.15-2               1.3M
-libseccomp2          2.3.3-4                 306K
-libssl1.1            1.1.1d-0 deb10u3        4M
-libsystemd0          241-7~deb10u3           767K
-libtasn1-6           4.13-3                  112K
-libudev1             241-7~deb10u3           258K
-libunistring2        0.9.10-1                1.6M
-libzstd1             1.3.8 dfsg-3            670K

Version differences:
PACKAGE                         IMAGE1 (k8s.gcr.io/debian-iptables-amd64:v11.0.3)        IMAGE2 (us.gcr.io/k8s-artifacts-prod/build-image/debian-iptables-amd64:v12.1.0)
-adduser                        3.115, 849K                                              3.118, 849K
-apt                            1.4.9, 3.5M                                              1.8.2, 4M
-base-files                     9.9 deb9u12, 333K                                        10.3 deb10u3, 340K
-base-passwd                    3.5.43, 229K                                             3.5.46, 232K
-conntrack                      1:1.4.4 snapshot20161117-5, 102K                         1:1.4.5-2, 103K
-coreutils                      8.26-3, 14.7M                                            8.30-3, 15.4M
-dash                           0.5.8-2.4, 204K                                          0.5.10.2-5, 212K
-debconf                        1.5.61, 558K                                             1.5.71, 520K
-debian-archive-keyring         2017.5 deb9u1, 148K                                      2019.1, 198K
-debianutils                    4.8.1.1, 213K                                            4.8.6.1, 226K
-diffutils                      1:3.5-3, 1.3M                                            1:3.7-3, 1.5M
-dpkg                           1.18.25, 6.6M                                            1.19.7, 6.5M
-ebtables                       2.0.10.4-3.5 b1, 351K                                    2.0.10.4 snapshot20181205-3, 265K
-findutils                      4.6.0 git+20161106-2, 1.8M                               4.6.0 git+20190209-2, 1.9M
-gpgv                           2.1.18-8~deb9u4, 721K                                    2.2.12-1 deb10u1, 837K
-grep                           2.27-2, 1.1M                                             3.3-1, 1014K
-gzip                           1.6-5 b1, 231K                                           1.9-3, 243K
-hostname                       3.18 b1, 47K                                             3.21, 54K
-init-system-helpers            1.48, 131K                                               1.56 nmu1, 133K
-ipset                          6.30-2, 140K                                             6.38-1.2, 150K
-iptables                       1.6.0 snapshot20161117-6, 1.5M                           1.8.3-2~bpo10 1, 2.5M
-kmod                           23-2, 220K                                               26-1, 237K
-libacl1                        2.2.52-3 b1, 62K                                         2.2.53-4, 70K
-libapt-pkg5.0                  1.4.9, 3M                                                1.8.2, 3.2M
-libattr1                       1:2.4.47-2 b2, 42K                                       1:2.4.48-4, 55K
-libaudit-common                1:2.6.7-2, 30K                                           1:2.8.4-3, 33K
-libaudit1                      1:2.6.7-2, 150K                                          1:2.8.4-3, 161K
-libbz2-1.0                     1.0.6-8.1, 96K                                           1.0.6-9.2~deb10u1, 104K
-libc-bin                       2.24-11 deb9u4, 3.3M                                     2.28-10, 3.4M
-libc6                          2.24-11 deb9u4, 10.4M                                    2.28-10, 12M
-libcap-ng0                     0.7.7-3 b1, 43K                                          0.7.9-2, 47K
-libdb5.3                       5.3.28-12 deb9u1, 1.8M                                   5.3.28 dfsg1-0.5, 1.8M
-libdebconfclient0              0.227, 67K                                               0.249, 72K
-libgcc1                        1:6.3.0-18 deb9u1, 108K                                  1:8.3.0-6, 116K
-libgcrypt20                    1.7.6-2 deb9u3, 1.2M                                     1.8.4-5, 1.3M
-libgpg-error0                  1.26-2, 572K                                             1.35-1, 175K
-libiptc0                       1.6.0 snapshot20161117-6, 88K                            1.8.3-2~bpo10 1, 98K
-libkmod2                       23-2, 116K                                               26-1, 129K
-liblz4-1                       0.0~r131-2 b1, 93K                                       1.8.3-1, 141K
-liblzma5                       5.2.2-1.2 b1, 339K                                       5.2.4-1, 263K
-libnetfilter-conntrack3        1.0.6-2, 134K                                            1.0.7-1, 138K
-libnettle6                     3.3-1 b2, 358K                                           3.4.1-1, 380K
-libnfnetlink0                  1.0.1-3, 61K                                             1.0.1-3 b1, 41K
-libpam-modules                 1.1.8-3.6, 874K                                          1.3.1-5, 1M
-libpam-modules-bin             1.1.8-3.6, 220K                                          1.3.1-5, 238K
-libpam-runtime                 1.1.8-3.6, 1016K                                         1.3.1-5, 1M
-libpam0g                       1.1.8-3.6, 229K                                          1.3.1-5, 244K
-libpcre3                       2:8.39-3, 668K                                           2:8.39-12, 673K
-libselinux1                    2.6-3 b3, 209K                                           2.8-1 b1, 194K
-libsemanage-common             2.6-2, 39K                                               2.8-2, 30K
-libsemanage1                   2.6-2, 291K                                              2.8-2, 301K
-libsepol1                      2.6-2, 653K                                              2.8-1, 724K
-libstdc++6                     6.3.0-18 deb9u1, 2M                                      8.3.0-6, 2M
-libxtables12                   1.6.0 snapshot20161117-6, 135K                           1.8.3-2~bpo10 1, 149K
-login                          1:4.4-4.1, 2.7M                                          1:4.5-1.1, 2.6M
-lsb-base                       9.20161125, 49K                                          10.2019051400, 49K
-netbase                        5.4, 44K                                                 5.6, 44K
-passwd                         1:4.4-4.1, 2.4M                                          1:4.5-1.1, 2.5M
-perl-base                      5.24.1-3 deb9u6, 7.4M                                    5.28.1-6, 9.9M
-sed                            4.4-1, 799K                                              4.7-1, 883K
-tar                            1.29b-1.1, 2.7M                                          1.30 dfsg-6, 2.8M
-zlib1g                         1:1.2.8.dfsg-5, 156K                                     1:1.2.11.dfsg-1, 174K
```

**Comment by tpepper** [comment]
Ahead of expected OWNERS lgtm/approve's marking as cherry-pick-approved so the set of merge requirements hopefully come together a little faster and we get back to being build-ready.

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by tallclair** [comment]
Sorry I didn't catch this before it merged, but previously our stance has been that we don't cherry-pick major version upgrades of the base images, namely the stretch->buster upgrade.

I'm not too worried about it for the control plane images that are built on debian-base, but I'm more concerned about debian-iptables based images, since I don't think we have sufficient test coverage to catch issues with an iptables upgrade.

**Comment by tallclair** [comment]
Note that we've continued to cut new versions of `debian-base:v1.0.*` (and cherry-picked them to earlier releases) even after v2.0.0 was released in v1.17.

**Comment by spiffxp** [comment]
@tallclair Should we revert this?

**Comment by tallclair** [comment]
spoke with @justaugustus on slack - since this has already been released in 1.16.10, our recommendation is that we flag it for some extra manual testing by kube-proxy maintainers, but don't revert it unless issues are found. But I'd be open to reverting it, if you have a proposal for how best to handle that.

**Comment by BenTheElder** [comment]
This upgrade should actually make iptables work in more environments,
notably multiple stable distros shipping nftables backed by default

I sympathize with not doing backporting major upgrades though.

What do we think we're not covering with iptables and CI? My main concern
there would probably be kube-proxy in IPVS

**Comment by oded-dd** [comment]
We do see issue with this version (1.16.10) when using fluent-bit as a daemonset unable to connect fluentd for some of the pods

**Pull request merged**
